### DISCUSSION TITLE
Cannot enable s3 transfer acceleration for bucket provided externally

### DISCUSSION THREAD

**Discussion by Holi0317** [serverless/serverless#12134] [Q&A]
Hi,

I am planning to manage the deployment bucket in terraform externally so we only have 1 deployment bucket instead of a dozen created by serverless on-the-fly. During the migration, I got the following error:

```
It is not possible to enable S3 Transfer Acceleration on an user provided bucket. In order to avoid this error, stop using `--aws-s3-accelerate` flag
```

Digging into serverless code brings me to [generate-core-template.js](https://github.com/serverless/serverless/blob/2aba07bc10a2e5c77ff7a657bbfe94678e082f0d/lib/plugins/aws/package/lib/generate-core-template.js), which brings me to deprecation warning of [S3_TRANSFER_ACCELERATION_ON_EXISTING_BUCKET] on sls v3.

[S3_TRANSFER_ACCELERATION_ON_EXISTING_BUCKET]: https://www.serverless.com/framework/docs/deprecations/#attempt-to-enable-s3-transfer-acceleration-on-provided-s3-buckets

The change is committed on https://github.com/serverless/serverless/commit/04b921acdc5fd486ffd10fe81fcb2243f37329db. Is there any reason behind this change? There is no reasoning written for this change in the changelog or migration guide right now.

We want to enable transfer acceleration because our ci runs on us-east-1 while our deployment region is in eu and ap. Our lambda zip is around 50MB so enabling transfer acceleration should help with the deployment time.

**Comment by thomas-hallinan** [comment]
I agree with Holi0317.  We have a large stack, and our combined size of all lambdas is about 500 Mb.

**Comment by pgrzesik** [comment]
Hey @Holi0317 - it's not possible to do it on an externally owned bucket because that flag also controls how the bucket is going to be created - if it's provisioned by serverless, proper settings for S3 acceleration are also set on the bucket.

**Reply by ksparakis** [reply]
Hey!
Tagging along here to see if I can understand this.

why are we unable to provision that bucket with the same setting on our own via terraform? it should only be that enable accelerated transfer right

**Comment by pmyjavec** [comment]
I upgraded an old project and for some reason the previous person who worked on the project specified the bucket name, after removing `deploymentBucket.name` the error went away.
